packages/components/src/templates/next/components/internal/SearchableTable/getFilteredItems.ts

```diff
@@ -0,0 +1,15 @@
+import type { SearchableTableProps } from "~/interfaces"
+
+interface GetFilteredItemsParams {
+  items: SearchableTableProps["items"]
+  searchValue: string
+}
+
+export const getFilteredItems = ({
+  items,
+  searchValue,
+}: GetFilteredItemsParams) => {
+  return items.filter((item) =>
+    item.join(" ").toLowerCase().includes(searchValue.toLowerCase()),
```

Comment:
nit/perf: do we want to precache each "row" before searching them to make searching slightly more performant?
I'd think the items will never change since it is static

```ts
const cache = items.map(item => ({ ...item, key: item.join(" ").toLowerCase() }))

const getFilteredItems = ({cache, searchValue}) => {
  return cache.filter(item => item.key.includes(searchValue.toLowerCase()))
}
```

Reply:
Split into server and client parts, with the server part doing this pre-cache step in 44298b012cbed4c09b6d65838834415abe98f1f6.